rosidl_generator_c/rosidl_generator_c/__init__.py

```diff
@@ -229,6 +229,18 @@ def escape_wstring(s):
     return escape_string(s)
 
 
+def escape_utf8(s):
+    result = ''
+    for c in s.encode('utf-8'):
+        # ASCII characters are encoded as-is
+        # Non-ASCII characters are encoded as UTF-8
+        if c < 0x80:
+            result += chr(c)
+        else:
+            result += '\\x%02x' % c
```

Comment:
nit. comments stays where the code is implemented?

```suggestion
        if c < 0x80:
            # ASCII characters are encoded as-is
            result += chr(c)
        else:
            # Non-ASCII characters are encoded as UTF-8
            result += '\\x%02x' % c
```